src/Microsoft.Dnx.Tooling/Utils/LockFileUtils.cs

```diff
@@ -63,9 +66,54 @@ public static LockFileLibrary CreateLockFileLibrary(LockFileLibrary previousLibr
             return lockFileLib;
         }
 
-        public static LockFileTargetLibrary CreateLockFileTargetLibrary(LockFileLibrary library, IPackage package, RestoreContext context, string correctedPackageName)
+        public static LockFileProjectLibrary CreateLockFileProjectLibrary(LockFileProjectLibrary previousLibrary,
+                                                                          Runtime.Project project,
+                                                                          Runtime.Project library)
         {
-            var lockFileLib = new LockFileTargetLibrary();
+            var result = new LockFileProjectLibrary()
+            {
+                Name = library.Name,
+                Version = library.Version
+            };
+
+            if (previousLibrary?.Name == library.Name && previousLibrary?.Version == library.Version)
+            {
+                result.Path = previousLibrary.Path;
+            }
+            else
+            {
+                result.Path = PathUtility.GetRelativePath(project.ProjectFilePath, library.ProjectFilePath, '/');
+            }
+
+            return result;
+        }
+
+        public static LockFileTargetLibrary CreateLockFileTargetLibrary(LockFileProjectLibrary library,
+                                                                        Runtime.Project projectInfo,
+                                                                        RestoreContext context)
+        {
+            var lockFileLib = new LockFileTargetLibrary
+            {
+                Name = library.Name,
+                Version = library.Version,
+                Type = "project"
+            };
+
+            var targetFramework = projectInfo.GetTargetFramework(context.FrameworkName);
+            var dependencies = projectInfo.Dependencies.Concat(targetFramework?.Dependencies ?? Enumerable.Empty<Runtime.LibraryDependency>());
+
+            lockFileLib.Dependencies = dependencies.Select(d => new PackageDependency(d.LibraryRange.Name, d.LibraryRange.VersionRange))
```

Comment:
No it doesn't. Should we put framework assemblies in project target library as well?